modules/swagger-codegen/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -241,6 +241,8 @@ class ObjectSerializer
                 $values[] = $this->deserialize($value, $subClass);
             }
             $deserialized = $values;
+        } elseif ($class === 'ByteArray') { // byte array
+            $deserialized = unpack('C*', (string)$data);
```

Comment:
Here is a 1-liner to restore the byte array: https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/php/SwaggerClient-php/tests/PetApiTest.php#L329

Agree that it's not straightforward but that's how PHP represents byte array.
Ref: http://stackoverflow.com/questions/885597/string-to-byte-array-in-php